### PR TITLE
chore: add steering site info

### LINKED ISSUE #49: Improve clarity on steering vector extraction site
It's currently not immediately clear where we are extracting steering vectors from. 

A simple way to do this might be to add the torch `module_name` to the `SteeringVector` class.

### PR DESCRIPTION
**Pull request by dtch1997**
Add module names to `SteeringPatchHandle` to enable easier tracking
Minor refactors to improve clarity

closes #49

### CODE REVIEW

File: steering_vectors/utils.py [hunk 1]
```diff
@@ -21,3 +25,38 @@ def batchify(
...
+        if major_version >= 2:
+            return "mps"
+    else:
+        return "cpu"
+
+    raise RuntimeError("Should not reach here!")
+
+
+@Singleton
+class DeviceManager:
```
Comment (chanind) [inline steering_vectors/utils.py:45]: It doesn't looks like this is ever being used. Why is this needed?

Comment (dtch1997) [reply steering_vectors/utils.py:45]: I'm using this in the examples

File: steering_vectors/steering_vector.py [hunk 2]
```diff
@@ -101,12 +105,13 @@ def patch_activations(
             module = get_module(model, layer_name)
             handle = module.register_forward_hook(
                 # create the hook via function call since python only creates new scopes on functions
-                _create_additive_hook(
+                _create_additive_hook_output(
                     target_activation.reshape(1, 1, -1), token_indices, operator
                 )
             )
+            module_names.append(layer_name + ".output")
```
Comment (chanind) [inline steering_vectors/steering_vector.py:112]: Where does this `.output` string come from? It seems like not all possible modules we could steer will have submodule called "output"?

File: steering_vectors/utils.py (hunk 1 above)
Comment (chanind) [reply steering_vectors/utils.py:45]: Why not just set the device explicitly in examples, e.g. put `DEVICE = 'cuda' if torch.cuda.is_available() else 'cpu'` at the top of the notebook? It feels odd to put example code into the public API of a library IMO

Comment (dtch1997) [reply steering_vectors/utils.py:45]: fair. I agree with that, but I also think this is just a useful pattern generally to control instantiation of new tensors. The nice thing about this is that you can nest multiple contexts, within which different defaults will be used. If it turns out we really don't need this outside the examples I'll remove it

File: steering_vectors/steering_vector.py (hunk 2 above)
Comment (dtch1997) [reply steering_vectors/steering_vector.py:112]: This is based on the observation that the `create_additive_hook` function modifies the output. I feel it's important to clarify that, as otherwise it's ambiguous whether we're steering at layer input vs layer output.

Comment (chanind) [reply steering_vectors/steering_vector.py:112]: But is there an actual module called "output" inside of every module we're applying steering? It looks like this string is also never being read out?

Comment (dtch1997) [reply steering_vectors/steering_vector.py:112]: There is not an actual module called "output". I agree it's misleading given the syntax because it implies the torch module has an 'output' module. open to suggestions on how to improve the naming convention. 

It's intended to be read by the user

File: steering_vectors/utils.py (hunk 1 above)
Comment (chanind) [reply steering_vectors/utils.py:45]: I can see how it would be used, but IMO if this is just for examples it's more confusing than just writing out the code to set the device in the notebook. This is a non-standard object we've created, and will require users to read the actual code to understand what it's doing. Personally, I'd think it's cleaner to just define this object in the notebook in that case, so then users don't need to look at our source code to understand what `DeviceManager.instance()` means, no? Plus it's relying on decorators and the singleton pattern, which is more things users need to understand rather than a simple `x.device = 'cuda'` which should be pretty clear

File: steering_vectors/steering_vector.py (hunk 2 above)
Comment (chanind) [reply steering_vectors/steering_vector.py:112]: Is the issue that the built-in torch `RemovableHandle` class doesn't include the name of the module it's corresponding to? If we think that's a problem, I'd vote to just put the `layer_name` in the list instead of appending `.output`. It still feels weird to me to add this field we're never reading, but it would at least be corresponding to the actual module then.

Comment (dtch1997) [reply steering_vectors/steering_vector.py:112]: Yeah, but in that case I still think it's a problem that it's not uniquely specified whether we're referring to the output or the input.

Comment (chanind) [reply steering_vectors/steering_vector.py:112]: I'm still confused what the use-case is here for adding this feature. It sounds like this is a form of documentation, to let the user know that the steering handle is editing the output of modules? If so, could this just be added to the documentation and explain that the steering handle is editing the output of modules? I'm not a huge fan of adding unused code as a form of documentation in the hope that the user inspects the object, but if you feel this is necessary, could the field you choose express this somehow? Something like `edit_module_outputs` or something?

File: steering_vectors/record_activations.py [hunk 3]
```diff
@@ -49,22 +64,24 @@ def record_activations(
...
         module = get_module(model, layer_name)
-        hook_fn = _create_read_hook(
+        hook_fn = _create_read_hook_output(
             layer_num, recorded_activations, clone_activations=clone_activations
         )
         hooks.append(module.register_forward_hook(hook_fn))
+        module_names.append(layer_name + ".output")
     try:
-        yield recorded_activations
+        handle = RecordHandle(recorded_activations, hooks, module_names)
```
Comment (chanind) [inline steering_vectors/record_activations.py:78]: This looks like a breaking change to a documented, public-facing method. Why is this change needed? It's OK to make breaking changes, but we should bump the major version of the library if we do this, and it should only be done if there's a clear use-case for it.
